integration/container/run_test.go

```diff
@@ -0,0 +1,65 @@
+package container
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/util"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/docker/docker/pkg/parsers/kernel"
+)
+
+func daemonStorageDriver(ctx context.Context, cli client.APIClient) (string, error) {
+	info, err := cli.Info(ctx)
+	if err != nil {
+		return "", err
+	}
+	return info.Driver, nil
+}
+
+// #34573
+func TestRunVolumeLeakage(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+
+	// Note whether the test is running on devicemapper (where the error being
+	// tested ocurred). We run it anyway just to be safe.
+	driver, _ := daemonStorageDriver(context.Background(), client)
+	if driver != "devicemapper" {
+		t.Logf("Active driver (%s) is not devicemapper, continuing anyway.", driver)
+	}
+
+	// On some older kernels we cannot guarantee that out fix for this bug
+	// actually works. So if there is an error the best we can do is just
+	// warn about it (if the kernel is older than 3.18, when
+	// torvalds/linux@8ed936b5671bfb33d89bc60bdcc7cf0470ba52fe was merged).
+	oldKernel := util.MustKernelCmp(kernel.VersionInfo{Kernel: 3, Major: 18, Minor: 0}) < 0
+	if oldKernel {
+		t.Logf("NOTE: This test may fail on your <3.18 kernels.")
```

Comment:
Some distributions may have backported the fix, so the kernel version alone isn't enough to know whether the test will work (but we don't hard fail on older kernels).